From: Iain Lane <iain@orangesquash.org.uk>
Date: Wed, 20 Sep 2017 15:01:41 +0100
Subject: manager: Disconnect from signals on the proxy when we're disposed

We're calling a callback on a proxy after it has been disposed. We
should make sure to disconnect from it when we go away.

https://bugzilla.gnome.org/show_bug.cgi?id=787897
---
 libnm/nm-manager.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libnm/nm-manager.c b/libnm/nm-manager.c
index d3df733..294b9de 100644
--- a/libnm/nm-manager.c
+++ b/libnm/nm-manager.c
@@ -192,8 +192,8 @@ init_dbus (NMObject *object)
 	                                property_info);
 
 	/* Permissions */
-	g_signal_connect (priv->proxy, "check-permissions",
-	                  G_CALLBACK (manager_recheck_permissions), object);
+	g_signal_connect_object (priv->proxy, "check-permissions",
+				 G_CALLBACK (manager_recheck_permissions), object, 0);
 }
 
 static NMClientPermission
